src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -620,12 +624,13 @@ private static string GetNextArgument(string arguments, ref int i)
         /// <summary>Gets the wait state for this Process object.</summary>
         private ProcessWaitState GetWaitState()
         {
-            if (_waitStateHolder == null)
+            ProcessWaitState waitState = _waitStateHolder?._state;
+            if (waitState == null)
             {
+                // This will throw
```

Comment:
In what situation?

Reply:
It always throws. This will be reverted too.